### PR TITLE
add runtime and packaging type info for mariner2 arm64

### PR DESCRIPTION
**Pull request by anamnavi**
# PR Summary

Add support for packaging for mariner 2 arm64 into an .rpm file. This is an fxdependent linux arm package.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: tools/packaging/packaging.psm1 [hunk 1]
```diff
@@ -50,7 +50,7 @@ function Start-PSPackage {
         [string]$Name = "powershell",
 
         # Ubuntu, CentOS, Fedora, macOS, and Windows packages are supported
-        [ValidateSet("msix", "deb", "osxpkg", "rpm", "rpm-fxdependent", "msi", "zip", "zip-pdb", "nupkg", "tar", "tar-arm", "tar-arm64", "tar-alpine", "fxdependent", "fxdependent-win-desktop", "min-size")]
+        [ValidateSet("msix", "deb", "osxpkg", "rpm", "rpm-fxdependent", "rpm-arm-fxdependent", "msi", "zip", "zip-pdb", "nupkg", "tar", "tar-arm", "tar-arm64", "tar-alpine", "fxdependent", "fxdependent-win-desktop", "min-size")]
```
Comment (anamnavi) [inline tools/packaging/packaging.psm1:53]:
```suggestion
        [ValidateSet("msix", "deb", "osxpkg", "rpm", "rpm-fxdependent", "rpm-arm64-fxdependent", "msi", "zip", "zip-pdb", "nupkg", "tar", "tar-arm", "tar-arm64", "tar-alpine", "fxdependent", "fxdependent-win-desktop", "min-size")]
```

File: build.psm1 [hunk 2]
```diff
@@ -970,9 +973,15 @@ function New-PSOptions {
 
     # Build the Output path
     if (!$Output) {
-        if ($Runtime -like 'fxdependent*') {
+        # if ($Runtime -like 'fxdependent*' -and -not $Runtime -like 'fxdependent*linux*') {
```
Comment (TravisEz13) [inline build.psm1:976]:
```suggestion
```

File: tools/packaging/packaging.psm1 [hunk 3]
```diff
@@ -50,7 +50,7 @@ function Start-PSPackage {
         [string]$Name = "powershell",
 
         # Ubuntu, CentOS, Fedora, macOS, and Windows packages are supported
-        [ValidateSet("msix", "deb", "osxpkg", "rpm", "rpm-fxdependent", "msi", "zip", "zip-pdb", "nupkg", "tar", "tar-arm", "tar-arm64", "tar-alpine", "fxdependent", "fxdependent-win-desktop", "min-size")]
+        [ValidateSet("msix", "deb", "osxpkg", "rpm", "rpm-fxdependent", "rpm-arm64-fxdependent", "msi", "zip", "zip-pdb", "nupkg", "tar", "tar-arm", "tar-arm64", "tar-alpine", "fxdependent", "fxdependent-win-desktop", "min-size")]
```
Comment (TravisEz13) [inline tools/packaging/packaging.psm1:53]:
```suggestion
        [ValidateSet("msix", "deb", "osxpkg", "rpm", "rpm-fxdependent", "rpm-fxdependent-arm64", "msi", "zip", "zip-pdb", "nupkg", "tar", "tar-arm", "tar-arm64", "tar-alpine", "fxdependent", "fxdependent-win-desktop", "min-size")]
```

File: tools/packaging/packaging.psm1 [hunk 4]
```diff
@@ -109,7 +109,10 @@ function Start-PSPackage {
             }
         } elseif ($Type.Count -eq 1 -and $Type[0] -eq "rpm-fxdependent") {
             New-PSOptions -Configuration "Release" -Runtime 'fxdependent-linux-x64' -WarningAction SilentlyContinue | ForEach-Object { $_.Runtime, $_.Configuration }
-        } else {
+        } elseif ($Type.Count -eq 1 -and $Type[0] -eq "rpm-arm64-fxdependent") {
```
Comment (TravisEz13) [inline tools/packaging/packaging.psm1:112]:
```suggestion
        } elseif ($Type.Count -eq 1 -and $Type[0] -eq "rpm-fxdependent-arm64") {
```

File: tools/packaging/packaging.psm1 [hunk 5]
```diff
@@ -633,6 +639,26 @@ function Start-PSPackage {
                     }
                 }
             }
+            'rpm-arm64-fxdependent' {
```
Comment (TravisEz13) [inline tools/packaging/packaging.psm1:642]:
```suggestion
            'rpm-fxdependent-arm64' {
```

File: tools/packaging/packaging.psm1 [hunk 6]
```diff
@@ -937,6 +968,21 @@ function New-UnixPackage {
...
+        #     $ParameterAttrHA = New-Object "System.Management.Automation.ParameterAttribute"
+        #     $ValidateSetAttrHA = New-Object "System.Management.Automation.ValidateSetAttribute" -ArgumentList "x86_64", "aarch64"
+        #     $AttributesHA = New-Object "System.Collections.ObjectModel.Collection``1[System.Attribute]"
+        #     $AttributesHA.Add($ParameterAttrHA) > $null
+        #     $AttributesHA.Add($ValidateSetAttrHA) > $null
+        #     $ParameterHA = New-Object "System.Management.Automation.RuntimeDefinedParameter" -ArgumentList ("HostArchitecture", [string], $AttributesHA)
+        #     $Dict = New-Object "System.Management.Automation.RuntimeDefinedParameterDictionary"
+        #     $Dict.Add("HostArchitecture", $ParameterHA) > $null
+        #     $Type = "rpm"
+        # }
```
Comment (TravisEz13) [inline tools/packaging/packaging.psm1:985]:
```suggestion
```

File: tools/packaging/packaging.psm1 [hunk 7]
```diff
@@ -1128,7 +1175,14 @@ function New-UnixPackage {
         # Setup package dependencies
         $Dependencies = @(Get-PackageDependencies @packageDependenciesParams)
 
-        $Arguments = Get-FpmArguments `
+        $Arguments = @()
+
+        # if ($HostArchitecture -and $Type -eq "rpm")
+        # {
+        #     $Arguments += @("-a", $HostArchitecture)
+        # }
```
Comment (TravisEz13) [inline tools/packaging/packaging.psm1:1183]:
```suggestion
```

File: tools/ci.psm1 [hunk 8]
```diff
@@ -784,7 +784,7 @@ function New-LinuxPackage
 
     # Only build packages for PowerShell/PowerShell repository
     # branches, not pull requests
-    $packages = @(Start-PSPackage @packageParams -SkipReleaseChecks -Type deb, rpm, tar)
+    $packages = @(Start-PSPackage @packageParams -SkipReleaseChecks -Type deb, rpm, rpm-arm64-fxdependent, tar)
```
Comment (TravisEz13) [inline tools/ci.psm1:787]:
```suggestion
    $packages = @(Start-PSPackage @packageParams -SkipReleaseChecks -Type deb, rpm, rpm-fxdependent-arm64, tar)
```

**Review by TravisEz13 — COMMENTED** [review]
Looks good, except a few commented code sections and I think we should be consistent about the order of how we put together the names of things.

Specifically, in other places, we put the architecture after fxdependent.

**Review by TravisEz13 — APPROVED** [review]

**Comment by TravisEz13** [comment]
@PowerShell/powershell-maintainers We will defer backporting to LTS until we have more data.

**Pull request merged**